Documentation/raspi-i2c.md

```diff
@@ -0,0 +1,170 @@
+
+The most simplest code you can build to use hardware I2C is the following (require C#9.0 to be activated):
+
+```csharp
+using System;
+using System.Device.I2c;
+
+Console.WriteLine("Hello I2C!");
+I2cDevice i2c = I2cDevice.Create(new I2cConnectionSettings(1, 0x12));
+i2c.WriteByte(0x42);
+var read = i2c.ReadByte();
+```
+
+In this case ```I2cDevice.Create(new I2cConnectionSettings(1, 0x12))``` will create an I2C device on the bus 1 (the default one) and with the device address 0x12.
+
+If you get an error message like this one, it means I2C has not been properly enabled:
+
+```text
+Hello I2C!
+Unhandled exception. System.IO.IOException: Error 2. Can not open I2C device file '/dev/i2c-1'.
+   at System.Device.I2c.UnixI2cDevice.Initialize()
+   at System.Device.I2c.UnixI2cDevice.WriteByte(Byte value)
+   at TestTest.Program.Main(String[] args) in C:\tmp\TestI2C\Program.cs:line 5
+Aborted
+```
+
+If you get an error message like the following one, it means that most likely you have a problem with cabling or your I2C address:
+
+```text
+Unhandled exception. System.IO.IOException: Error 121 performing I2C data transfer.
+   at System.Device.I2c.UnixI2cDevice.ReadWriteInterfaceTransfer(Byte* writeBuffer, Byte* readBuffer, Int32 writeBufferLength, Int32 readBufferLength)
+   at System.Device.I2c.UnixI2cDevice.Transfer(Byte* writeBuffer, Byte* readBuffer, Int32 writeBufferLength, Int32 readBufferLength)
+   at System.Device.I2c.UnixI2cDevice.WriteByte(Byte value)
+   at TestTest.Program.Main(String[] args) in C:\tmp\TestI2C\Program.cs:line 14
+Aborted
+```
+
+**Note**: In rare cases, you might see the above exception during normal operation of a device as well. Adding retries might solve the issue.
+
+See at the end if you want to write your own I2C scanner and find the correct device address.
+
+## Enabling I2C
+
+In most of the cases, it is easy and straight forward to enable I2C for your Raspberry Pi. The basic case can be [found here](https://www.raspberrypi-spy.co.uk/2014/11/enabling-the-i2c-interface-on-the-raspberry-pi/).
+
+Sometimes, you have I2C devices which can't support fast mode and the speed needs to be adjusted. In case you want to change the speed of the line, you'll need to add into the `/boot/config.txt` file:
+
+```bash
+sudo nano /boot/config.txt
+```
+
+Add the line which will change the speed from 100_000 (default speed) to 10_000:
+
+```text
+dtparam=i2c_arm_baudrate=10000
+```
+
+Save the file with `ctrl + x` then `Y` then `enter`
+
+Then reboot:
+
+```bash
+sudo reboot
+```
+
+More information on dtoverlay and how to select specific elements of I2C buses are [available here](https://github.com/raspberrypi/firmware/blob/bff705fffe59ad3eea33999beb29c3f26408de40/boot/overlays/README#L1387). 2 busses are available on any Raspberry Pi. More can be available on specific models like those base out of BCM2711 such as Raspberry Pi 4.
+
+### Enabling I2C with advance parameters
+
+The general pattern looks like the following:
```

Comment:
you should also mention you can do `sudo raspi-config nonint do_i2c 0` on raspbian

Reply:
Added. 1 is the default, so changed for 1 with a note